### PR TITLE
Doc: Add initial documentation to the simulator

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Add initial documentation to the simulator
## Impact
Users will able use the simulator
## Testing
N/A

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: Documentation/platforms/sim/sim/index.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+===
+SIM
+===
+
+It is possible to run NuttX in a simulator called ``sim``, but currently it only
+supports Linux as host platform.
```
Comment (Ouss4) [inline Documentation/platforms/sim/sim/index.rst:6]: Is this (still) true? I think it was used on both macOS and Windows.

Comment (acassis) [reply Documentation/platforms/sim/sim/index.rst:6]: It was developed for Linux, all main features depends on Linux subsystems: BT, I2C, SPI, etc

Comment (xiaoxiang781216) [reply Documentation/platforms/sim/sim/index.rst:6]: But, other feature(especially network and hostfs) work on macOS.

Comment (acassis) [reply Documentation/platforms/sim/sim/index.rst]: Ok, fixed!

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**